src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -417,6 +422,61 @@ public void AddHostObject(string name, FormulaType type, Func<IServiceProvider,
                 displayName: default);
 
             _variables.Add(hostDName, info);
-        }
+        }
+
+        /// <summary>
+        /// Adds a named type that can be referenced in expression.
+        /// </summary>
+        /// <param name="typeName">Name of the type to be added into Symbol table.</param>
+        /// <param name="type">Type associated with the name.</param>
+        public void AddType(DName typeName, FormulaType type)

```

Comment:
>DName typeName [](http://example.com/codeflow?start=28&length=14)

All other apis uses string vs DName in symbol table, should it be the same? 
Also we should validate the DName, see `ValidateName` method.

Reply:
since validity is checked on the DName constructor , do we need to validate again ?